bin/zField

```diff
@@ -0,0 +1,271 @@
+    elif database == 'KiDSDR4':
+        retriever=catalogRetriever.KiDSDR4Retriever
+        zPriorMin=0.05
+        zPriorMax=1.5
+        passbandSet='KiDS-VIKING'
+        zDebias=0.02    # From testing against spec-zs (don't know yet why there is a mean offset)
+        retrieverOptions={'maxMagError': maxMagError}
+    elif database == 'ATLASDR4':
+        retriever=catalogRetriever.ATLASDR4Retriever
+        zPriorMin=0.05
+        zPriorMax=0.8
+        passbandSet='KiDS-VIKING'
+        retrieverOptions={'maxMagError': maxMagError}
+    elif database == 'CFHTLenS':
+        retriever=catalogRetriever.CFHTLenSRetriever
+        zPriorMin=0.05
+        zPriorMax=1.5
+        retrieverOptions={'maxMagError': maxMagError}
+    elif database == 'DECaLS':
+        # For DECaLS, need the bricks files that define survey on the sky
+        # These were previously included in zCluster, but now we fetch over web and cache
+        bricksCacheDir=catalogRetriever.CACHE_DIR
+        catalogRetriever.makeCacheDir()
+        bricksPath=bricksCacheDir+os.path.sep+"survey-bricks.fits.gz"
+        if os.path.exists(bricksPath) == False:
+            print("... fetching and caching DECaLS survey-bricks.fits.gz ...")
+            urllib.request.urlretrieve("http://portal.nersc.gov/project/cosmo/data/legacysurvey/dr8/survey-bricks.fits.gz", bricksPath)
+        bricksDR8Path=bricksCacheDir+os.path.sep+"survey-bricks-dr8-south.fits.gz"
+        if os.path.exists(bricksDR8Path) == False:
+            print("... fetching and caching DECaLS survey-bricks-dr8-south.fits.gz ...")
+            urllib.request.urlretrieve("https://portal.nersc.gov/project/cosmo/data/legacysurvey/dr8/south/survey-bricks-dr8-south.fits.gz", bricksDR8Path)
+        bricksTab=atpy.Table().read(bricksPath)
+        DR8Tab=atpy.Table().read(bricksDR8Path)
+        DR8Tab.rename_column("brickname", "BRICKNAME")
+        retriever=catalogRetriever.DECaLSRetriever
+        zPriorMin=0.05
+        zPriorMax=1.5
+        #maxMagError=0.2
+        passbandSet='DECaLS'
+        # Zero point offsets had no effect on correcting the bias
+        #ZPOffsets=np.array([-0.00878495, -0.019883, 0.01998477, -0.03047624, 0.00860256])
+        #ZPOffsets=np.array([-0.00380799, -0.01845335, 0.01706214, -0.0291939, 0.01454793])
+        #ZPOffsets=np.array([-0.07441184, -0.04284442, 0.01009525, 0.01008043, 0.05755784]) 
+        #ZPOffsets=np.array([0.02271222, -0.05051711, -0.02465597, -0.00406835, 0.05406105])
+        zDebias=0.02    # From testing against spec-zs (don't know yet why there is a mean offset)
+        retrieverOptions={'maxMagError': maxMagError, 'bricksTab': bricksTab, 'DR8Tab': DR8Tab, 'altCacheDir': "DECaLSCache"}
+    elif database == 'CFHTDeep':
+        retriever=catalogRetriever.CFHTDeepRetriever
+    elif database == 'CFHTWide':
+        retriever=catalogRetriever.CFHTWideRetriever
+        zPriorMin=0.05
+        zPriorMax=1.5
+        
+    if passbandSet == 'SDSS+Ks':
```

Comment:
Instead of this block here, get the names of the bands from PhotoRedshiftEngine.bands.